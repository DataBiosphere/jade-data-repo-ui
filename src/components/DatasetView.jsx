import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import { Link, NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

import { getDatasets } from 'actions/index';
import JadeTable from './table/JadeTable';
import AddSVG from '../../assets/media/icons/plus-circle-solid.svg';

const styles = theme => ({
  wrapper: {
    padding: theme.spacing.unit * 4,
    margin: theme.spacing.unit * 4,
  },
  title: {
    color: theme.palette.primary.main,
    fontSize: '54px',
    lineHeight: '66px',
    paddingBottom: theme.spacing.unit * 8,
  },
  header: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'row',
    fontSize: '18px',
    fontWeight: '600',
    paddingTop: '30px',
  },
  plusButton: {
    height: '30px',
    fill: theme.palette.primary.main,
    marginLeft: '10px',
    width: '30px',
  },
});

class DatasetView extends React.PureComponent {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    datasets: PropTypes.object,
    dispatch: PropTypes.func.isRequired,
  };

  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(getDatasets());
  }

  render() {
    const { classes, datasets } = this.props;
    const columns = [
      {
        label: 'Dataset Name',
        property: 'name',
        render: row => <Link to={`/dataset/${row.id}`}>{row.name}</Link>,
      },
      {
        label: 'Description',
        property: 'description',
      },
      {
        label: 'Last changed',
        property: 'modifiedDate',
        render: row => moment(row.modifiedDate).fromNow(),
      },
      {
        label: 'Date created',
        property: 'createdDate',
        render: row => moment(row.createdDate).fromNow(),
      },
    ];
    return (
      <div id="datasets" className={classes.wrapper}>
        <div className={classes.title}>Datasets</div>
        <p> Datasets make access control simple </p>
        <div className={classes.header}>
          DATASETS
          <NavLink to="/datasets/create">
            <AddSVG className={classes.plusButton} />
          </NavLink>
        </div>
        <div>
          {datasets && datasets.datasets && (
            <JadeTable columns={columns} rows={datasets.datasets} />
          )}
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return { datasets: state.datasets };
}

export default connect(mapStateToProps)(withStyles(styles)(DatasetView));
