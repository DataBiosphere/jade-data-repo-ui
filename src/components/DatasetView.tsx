import React, { Dispatch } from 'react';
import { connect } from 'react-redux';
import { Action } from 'redux';
import { WithStyles, withStyles } from '@mui/styles';
import { getDatasets, addDatasetPolicyMember } from 'actions/index';
import { DatasetSummaryModel } from 'generated/tdr';
import { CustomTheme } from '@mui/material';
import { TdrState } from 'reducers';
import { OrderDirectionOptions } from 'reducers/query';
import DatasetTable from './table/DatasetTable';

import { DatasetRoles } from '../constants';

const styles = (theme: CustomTheme) => ({
  wrapper: {
    display: 'flex',
    justifyContent: 'center',
    marginTop: '1em',
  },
  width: {
    ...theme.mixins.containerWidth,
  },
  title: {
    color: theme.palette.primary.main,
    fontSize: '54px',
    lineHeight: '66px',
    paddingBottom: theme.spacing(8),
  },
});

interface IProps extends WithStyles<typeof styles> {
  datasets: Array<DatasetSummaryModel>;
  datasetRoleMaps: { [key: string]: Array<string> };
  datasetsCount: number;
  dispatch: Dispatch<Action>;
  filteredDatasetsCount: number;
  loading: boolean;
  searchString: string;
  refreshCnt: number;
  user: string;
}

const DatasetView = withStyles(styles)(
  ({
    classes,
    datasets,
    datasetRoleMaps,
    datasetsCount,
    dispatch,
    filteredDatasetsCount,
    loading,
    searchString,
    refreshCnt,
    user,
  }: IProps) => {
    const handleFilterDatasets = (
      limit: number,
      offset: number,
      sort: string,
      sortDirection: OrderDirectionOptions,
      search: string,
    ) => {
      dispatch(getDatasets(limit, offset, sort, sortDirection, search));
    };

    const handleMakeSteward = (datasetId: string) => {
      dispatch(addDatasetPolicyMember(datasetId, user, DatasetRoles.STEWARD));
    };

    return (
      <div className={classes.wrapper}>
        <div className={classes.width}>
          <div>
            {datasets && (
              <DatasetTable
                datasets={datasets}
                datasetRoleMaps={datasetRoleMaps}
                datasetsCount={datasetsCount}
                handleFilterDatasets={handleFilterDatasets}
                handleMakeSteward={handleMakeSteward}
                filteredDatasetsCount={filteredDatasetsCount}
                searchString={searchString}
                loading={loading}
                refreshCnt={refreshCnt}
              />
            )}
          </div>
        </div>
      </div>
    );
  },
);

function mapStateToProps(state: TdrState) {
  return {
    datasets: state.datasets.datasets,
    datasetRoleMaps: state.datasets.datasetRoleMaps,
    datasetsCount: state.datasets.datasetsCount,
    filteredDatasetsCount: state.datasets.filteredDatasetsCount,
    features: state.user.features,
    loading: state.datasets.loading,
    user: state.user.email,
    refreshCnt: state.datasets.refreshCnt,
  };
}

export default connect(mapStateToProps)(DatasetView);
