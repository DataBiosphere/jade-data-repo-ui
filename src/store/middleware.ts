import createSagaMiddleware from 'redux-saga';

export const sagaMiddleware = createSagaMiddleware();

const middleware = [sagaMiddleware];

/* istanbul ignore next */
if (process.env.NODE_ENV === 'development') {
  //eslint-disable-next-line  @typescript-eslint/no-var-requires
  const { createLogger } = require('redux-logger');
  //eslint-disable-next-line  @typescript-eslint/no-var-requires
  const invariant = require('redux-immutable-state-invariant').default;

  middleware.push(invariant());
  middleware.push(createLogger({ collapsed: true }));
}

export default middleware;
