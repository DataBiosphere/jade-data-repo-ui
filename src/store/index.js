import { applyMiddleware, createStore, compose, combineReducers } from 'redux';
import { createForms } from 'react-redux-form';

import rootSaga from 'sagas/index';
import rootReducer from 'reducers/index';

import middleware, { sagaMiddleware } from './middleware';

const initialDatasetState = {
  name: '',
  description: '',
  readers: [],
  study: '',
};

const reducer = combineReducers({
  ...rootReducer,
  ...createForms({ dataset: initialDatasetState }),
});

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/* istanbul ignore next */
const configStore = (initialState = {}) => {
  const store = createStore(reducer, initialState, composeEnhancer(applyMiddleware(...middleware)));

  sagaMiddleware.run(rootSaga);

  if (module.hot) {
    module.hot.accept('reducers', () => {
      store.replaceReducer(require('reducers/index').default);
    });
  }

  return {
    store,
  };
};

const { store } = configStore();

global.store = store;

export { store };
