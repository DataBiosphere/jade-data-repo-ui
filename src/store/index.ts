import { applyMiddleware, combineReducers, compose, createStore, Store } from 'redux';
import { connectRouter } from 'connected-react-router';

import rootSaga from 'sagas/index';
import rootReducer, { initialTdrState, TdrState } from 'reducers/index';
import history from 'modules/hist';
import { History } from 'history';
import middleware, { sagaMiddleware } from './middleware';

const reducer = (hist: History) =>
  combineReducers({
    router: connectRouter(hist),
    ...rootReducer,
  });

// @ts-ignore
const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

/* istanbul ignore next */
const configStore = (initialState: TdrState = initialTdrState): Store<TdrState> => {
  const store = createStore(
    reducer(history),
    initialState,
    composeEnhancer(applyMiddleware(...middleware)),
  );

  sagaMiddleware.run(rootSaga);

  // @ts-ignore
  if (module.hot) {
    // @ts-ignore
    module.hot.accept('reducers', () => {
      store.replaceReducer(require('reducers/index').default);
    });
  }

  return store;
};

const store = configStore();

// @ts-ignore
global.store = store;

export { store };
