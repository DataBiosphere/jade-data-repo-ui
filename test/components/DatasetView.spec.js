import React from 'react';
import { Provider } from 'react-redux';

import { store } from 'store/index';
import DatasetView from 'components/DatasetView';

const props = {
  classes: {},
};

function setup(ownProps = props) {
  return mount(
    <Provider store={store}>
      <DatasetView {...ownProps} />
    </Provider>,
  );
}

describe('DatasetView', () => {
  const wrapper = setup();

  it('hey look a datasets view', () => {
    wrapper.find('#datasets').simulate('click');
  });
});
