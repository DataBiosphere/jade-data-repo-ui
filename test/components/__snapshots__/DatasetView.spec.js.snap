// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetView should render all zones 1`] = `
<Connect(DatasetView)
  app={
    Object {
      "alerts": Array [],
    }
  }
  dispatch={[MockFunction]}
/>
`;
